src/uu/install/src/install.rs

```diff
@@ -561,3 +568,75 @@ fn copy(from: &PathBuf, to: &PathBuf, b: &Behavior) -> Result<(), ()> {
 
     Ok(())
 }
+
+/// Return true if a file is necessary to copy.
+///
+/// # Parameters
+///
+/// _from_ and _to_, if existent, must be non-directories.
+///
+/// # Errors
+///
+/// Crashes the program if a nonexistent owner or group is specified in _b_.
+///
+fn need_copy(from: &PathBuf, to: &PathBuf, b: &Behavior) -> bool {
+    // setuid || setgid || sticky
+    let extra_mode: u32 = 0o7000;
+
+    let from_meta = match fs::metadata(from) {
+        Ok(meta) => meta,
+        Err(_) => return true,
+    };
+    let to_meta = match fs::metadata(to) {
+        Ok(meta) => meta,
+        Err(_) => return true,
+    };
+
+    if b.specified_mode.unwrap_or(0) & extra_mode != 0
+        || from_meta.mode() & extra_mode != 0
+        || to_meta.mode() & extra_mode != 0
+    {
+        return true;
+    }
+
+    if !from_meta.is_file() || !to_meta.is_file() {
+        return true;
```

Comment:
ditto

Reply:
This one's tricky. I added a (broken, ignored) test for this with a comment inside describing where I'm stuck.